Sources/Common/Module/ModuleTopLevelConfigOptions.swift

```diff
@@ -0,0 +1,2 @@
+public protocol ModuleTopLevelConfig {
```

Comment:
I currently don't see a use for this protocol in this PR. Perhaps it goes to a different PR?

Reply:
I was actually using this in upcoming PRs, but I no longer find this helpful, so removed this for now.